### DISCUSSION TITLE
Any downside to using optimized/optimized-turbo?

### DISCUSSION THREAD

**Discussion by thedyze** [hlky/stable-diffusion-webui#598] [General]
Any degradation in generation or image quality when using these parameters?

Also any information on the difference between optimized and optimized-turbo appreciated.

**Comment by KyuubiYoru** [comment]
Less vram use with the cost of longer processing time.

But not sure if there is a difference in quality, you could generate images with the same parameters and compare them.

**Comment by TheLastBen** [comment]
The optimized-turbo version is as fast as the regular version, the quality is the same, but VRAM and RAM are very active and stressed during the rendering, so if you have a lot of memory and want to protect your hardware, don't use the optimized version

**Comment by EliEron** [comment]
Since I was a bit curious about this myself I decided to do some tests. The result of my tests showed that the optimize mode does change the result slightly. Though I wouldn't say it degrades the result, it just make it very slightly different. Meaning the same seed with the same parameters won't result in literally the same image.

The differences fell in two camps, images where there were minor but noticeable differences, and images where there were differences but they were practically unnoticeable.

Here is an example where there actually are notable differences between the normal mode and the optimize mode:

Normal:
![normal-1](https://user-images.githubusercontent.com/22575945/188296846-1824668a-f5b7-45b7-b0a9-581a681d956f.png)

Optimized:
![optimized-1](https://user-images.githubusercontent.com/22575945/188296850-9707df91-a671-45b3-95df-1627ebea168d.png)
(Open both in separate tabs and switch between them to see the differences, they are mostly minor, but there are quite a few)

And here is an example where there is technically a difference but it is practically unnoticeable:

![normal-2](https://user-images.githubusercontent.com/22575945/188297308-48f4523c-4745-4d5a-a3ab-0ed027029858.png)
![optimized-2](https://user-images.githubusercontent.com/22575945/188297313-a1acc362-284e-42da-9a27-38fec79e572e.png)

So in conclusion, there are differences, but they really aren't something I'd worry about. The image variance you get just from using a different sampler or running for a different amount of steps far outweighs the difference between the normal and optimized versions.

**Reply by thedyze** [reply]
Cool!
Does generating an image with the same seed and parameters always produce an exactly identical image otherwise? (i.e two images with optimized for instance)

**Reply by EliEron** [reply]
Yes, if you use the same prompt, seed and parameters (CFG, sampler, steps) then you will always get exactly the same image returned by the model. This is true for the optimized mode as well. It's only when compared to the normal version that a difference is seen with the same parameters.

It's also worth noting that I did not discover any difference whatsoever between optimized and optimized-turbo in terms of produced images. Though the speed difference between them is truly immense. So if you have the VRAM I really can't recommend enough to use optimized-turbo over plain optimized.

The recent attention.py optimization that is currently open as a PR should reduce the VRAM usage even further. So if you don't have enough VRAM right now you might have it once that is merged.
